lib/basket.rb

```diff
@@ -0,0 +1,35 @@
+require_relative 'menu'
+
+class Basket
+
+    def initialize(menu = Menu.new.menu)
+      @menu = menu
+      @basket = []
+      @order_total = 0
+    end
+
+    def basket
+      @basket
+    end
+
+    def order_total
+      @order_total
+    end
+
+    def add_to_basket(item, quantity)
+      if valid_item?(item)
```

Comment:
I like the idea that you're checking if item is valid.
For clarity I would put raise statement first before if statement